controllers/service/service_controller.go

```diff
@@ -88,6 +90,13 @@ func (r *serviceReconciler) reconcile(ctx context.Context, req ctrl.Request) err
 		return client.IgnoreNotFound(err)
 	}
 	if !svc.DeletionTimestamp.IsZero() {
+		deletionProtectionEnabled, deletionProtectionSpecified, err := r.getDeletionProtectionViaAnnotation(*svc)
+		if err != nil {
+			return err
+		}
+		if deletionProtectionSpecified && deletionProtectionEnabled {
```

Comment:
deleteProtectionEnabled true would always imply deleteProtectionSpecified, so no need to check for both.

Reply:
Changed.